kong/db/migrations/core/003_100_to_110.lua

```diff
@@ -309,26 +309,30 @@ return {
   },
 
   cassandra = {
-    up = [[
-      ALTER TABLE plugins ADD protocols set<text>;
-
-      ALTER TABLE services ADD tags set<text>;
-      ALTER TABLE routes ADD tags set<text>;
-      ALTER TABLE certificates ADD tags set<text>;
-      ALTER TABLE snis ADD tags set<text>;
-      ALTER TABLE consumers ADD tags set<text>;
-      ALTER TABLE plugins ADD tags set<text>;
-      ALTER TABLE upstreams ADD tags set<text>;
-      ALTER TABLE targets ADD tags set<text>;
-
-      CREATE TABLE IF NOT EXISTS tags (
-        tag               text,
-        entity_name       text,
-        entity_id         text,
-        other_tags        set<text>,
-        PRIMARY KEY       ((tag), entity_name, entity_id)
-      );
-
-    ]],
+    up_f = function(connector)
+      -- Ignore errors
+      connector:query([[
+        ALTER TABLE plugins ADD protocols set<text>;
+
+        ALTER TABLE services ADD tags set<text>;
+        ALTER TABLE routes ADD tags set<text>;
+        ALTER TABLE certificates ADD tags set<text>;
+        ALTER TABLE snis ADD tags set<text>;
+        ALTER TABLE consumers ADD tags set<text>;
+        ALTER TABLE plugins ADD tags set<text>;
+        ALTER TABLE upstreams ADD tags set<text>;
+        ALTER TABLE targets ADD tags set<text>;
+      ]])
```

Comment:
It does not feel safe to me to blindly ignore *all* errors in a process as delicate as these automated C* migrations... Could we do something to only introduce this behavior when connecting to C* 4.x, and to only ignore specifically cherry-picked errors such as the one you observed @mikefero? Something like:
```lua
local res, err = connector:query(...)
if connector.major_version >= 4 and res == nil then
  if not [match err] then
    error(err)
  end
end
```
Doing so would also be helpful in that it would start a record of Kong/C* 4.x compatibility issues and possible roadmap blockers.

Reply:
I think we have same issue with other versions too, but currently we ignore errors by error message and each cassandra version changes them, @mikefero can you check this:
https://github.com/Kong/kong/blob/master/kong/db/strategies/cassandra/connector.lua#L1019

Aka just add new strings there and it should be fine?

Reply:
Thanks @bungle, that makes a lot more sense because I have seen the other ALTER commands ignored for the different versions of C*.